### DISCUSSION TITLE
TweeGoMatic extension for Tweego

### DISCUSSION THREAD

**Discussion by TheBlueJester** [tmedwards/tweego#42] [Show and tell]
Howdy Everyone!

I wanted to share a fork of Tweego that I spent today working on that I've named TweeGoMatic.
https://github.com/TheBlueJester/TweeGoMatic

It's nothing ground-breaking. 
I've just recompiled tweego using the latest version of Go, updated the Story Format files to their current versions and thrown in a few Batch files for Windows that automate the use of Tweego.

I don't know if the author of this software still check in on this Repo after so long but if they do, please let me know if there are any problems with what I've done. This is my first github fork and I'm a little worried about etiquette ect.

**Comment by tmedwards** [comment]
I do check the repo.  Tweego isn't abandoned by any means, I've simply not had the ability to work on it in a while.

As far as TweeGoMatic goes, I'm not sure I see the point.  No offense intended.

IMO, it's better if developers learn either how to use their tools, Tweego in this case, or the tools/packages that use those tools to provide extra value. For example:

* VSCode, that allows various kinds of scripting to automate builds, among other uses.
* Project skeletons, like `tweego-setup`, that automate builds and provide useful scaffolding.

TweeGoMatic clearly is attempting to be in the latter group, a value add, but I think it's missing the mark.

Anyway.  I don't have a problem with TweeGoMatic in general, so knock yourself out.  That said, I am a little put off by you including Tweego's source within TweeGoMatic when it's unnecessary, but I'm not going to make a mountain out of that molehill.

**Reply by TheBlueJester** [reply]
Thanks for taking the time to reply!

I don't mind if no one sees any value in my little project. It was great to use GitHub for the first time and the result is going to be useful for me even if it's not for anyone else.

I'll try and take your criticism on board, I'll be more careful with trivial projects and putting them online and I'll create something more worthwhile in the future I'm sure of it.

In the value-add vein I am now looking at linking into twee language tools extension more directly via its own hook for adding tasks. As suggested by someone on Twine discord server.

I was operating under the belief that when one forks it's normal / expected for the entire source code to exist in the fork and not just files that have been changed or edited. But clearly I'm mistaken, I've never used GitHub for anything like this before. 

Sorry for the slight against you and other negative feelings about the copied code. I'm on my phone currently so I'm not sure I can delete it but I will in the morning if not. 

I'll only host new and edited/updated files from now on 😊.

**Reply by tmedwards** [reply]
> I was operating under the belief that when one forks it's normal / expected for the entire source code to exist in the fork and not just files that have been changed or edited.

You'd be correct if TweeGoMatic were an actual fork of Tweego, but it's not at this point, and maybe never.  Adding value add content, separate from the actual source code, does not a true fork make.

Calling one project a *fork* of another means that, at the very least, you're interested in, and possibly committed to, making changes to the source code.

As an example, `tweego-setup` does not include Tweego's source code, nor does it claim to be a fork.  It's definitely a value add project for Tweego though.

**Reply by tmedwards** [reply]
To be clear, while it's off putting to me on a few different levels, I'm not asking you to remove Tweego's source code from TweeGoMatic.  I simply don't see the need for it in your project at this point.

What I'm trying to say is that at this point TweeGoMatic isn't a natural fork of Tweego, it's just some extra non-source files.  In that case, it makes more sense to simply be its own project.

Beyond that.  If you ever do want to make changes to Tweego's source code, for whatever reason, then I'd certainly prefer you go though me to possibly get said changes included into Tweego proper rather than living on some fork.

That said, if I refuse to or simply cannot make the changes for any reason, then truly forking Tweego could certainly make sense then.

**Comment by TheBlueJester** [comment]
Howdy!

Thanks for all the information and teaching you've done here :)
I've taken it on board as best I can and taken actions to remedy raised issues.

I don't believe any more replies are necessary, or at least I hope.
I'll delete this discussion thread sometime in the near future as I think it's probably unnecessary and probably annoying/confusing for random users.

Have a great day!
